lib/minitest/reporters/base_reporter.rb

```diff
@@ -85,12 +85,13 @@ def print(*args)
         io.print(*args)
       end
 
-      def print_info(e, name=true)
-        print "#{e.exception.class.to_s}: " if name
+      def print_info(e, error=true)
         e.message.each_line { |line| print_with_info_padding(line) }
 
-        trace = filter_backtrace(e.backtrace)
-        trace.each { |line| print_with_info_padding(line) }
+        unless error
+          trace = filter_backtrace(e.backtrace)
+          trace.each { |line| print_with_info_padding(line) }
+        end
```

Comment:
> After the change it always prints name and prints backtraces only if name/error is true.
Imho it is too much to fix problem with particular exception (and it is unclear why the new code will work with other exceptions).

That's not right. Minitest wraps all types of exceptions in `UnexpectedError`. See [these lines](https://github.com/seattlerb/minitest/blob/e9752484d624950bb3af4e96fde118d921002dd2/lib/minitest/test.rb#L204-L212). And `UnexpectedError`'s `message` method already contains the backtrace. Hence the change here basically ensures that when an exception happens, this prints name, message and backtrace only once.

`print_name` for that parameter's name sounds good